conformance-tests/storage/retry/Dockerfile

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+FROM python:3.8@sha256:ff8ae88abfe6c829c188cc54737139f7c583279009ba34c96034e0e66053ac3f
```

Comment:
Do you really want to pin this to the hash? I think we can expect the upstream won't introduce any changes other than security patches.

Not a strong opinion though

Reply:
Some folks I've worked within the past have asked for this to try and ensure some level or reproducibility/traceability for the built images. If we don't have a similar requirement, I can relax this to just the `3.8` tag.

Reply:
ok